clang/lib/Sema/SemaDeclAttr.cpp

```diff
@@ -3171,18 +3191,38 @@ static void handleSubGroupSize(Sema &S, Decl *D, const ParsedAttr &AL) {
 }
 
 // Handles num_simd_work_items.
-static void handleNumSimdWorkItemsAttr(Sema &S, Decl *D, const ParsedAttr &A) {
+static void handleNumSimdWorkItemsAttr(Sema &S, Decl *D, const ParsedAttr &AL) {
   if (D->isInvalidDecl())
     return;
 
-  Expr *E = A.getArgAsExpr(0);
+  Expr *E = AL.getArgAsExpr(0);
 
   if (D->getAttr<SYCLIntelNumSimdWorkItemsAttr>())
-    S.Diag(A.getLoc(), diag::warn_duplicate_attribute) << A;
+    S.Diag(AL.getLoc(), diag::warn_duplicate_attribute) << AL;
 
-  S.CheckDeprecatedSYCLAttributeSpelling(A);
+  S.CheckDeprecatedSYCLAttributeSpelling(AL);
+
+  if (const auto *A = D->getAttr<ReqdWorkGroupSizeAttr>()) {
+    ASTContext &Ctx = S.getASTContext();
+
+    if (!E->isValueDependent()) {
+      Optional<llvm::APSInt> NumSimdWorkItems = E->getIntegerConstantExpr(Ctx);
```

Comment:
I don't think we've validated that this value isn't `0` yet and so the use of `%` below could crash in that case.

Reply:
I have validated the value that is not zero here to avoid crash and adds test for it.